src/box/lua/net_box.c

```diff
@@ -2673,10 +2673,15 @@ netbox_worker_f(va_list ap)
 		if (transport->state == NETBOX_CLOSED)
 			break;
 		netbox_transport_set_error(transport);
-		transport->state = (reconnect_after > 0 ?
+		/*
+		 * gh-7448: if the fiber is cancelled we should set the
+		 * NETBOX_ERROR state as no reconnection will be done anyway.
+		 */
```

Comment:
I think that a net.box worker fiber, like any other system fiber, should ignore `fiber.cancel` coming from the Lua world. The only way to break a connection from Lua should be by calling `conn.stop`. To achieve that, you could mark the net.box worker fiber as non-cancellable, see `fiber_set_cancellable`, but we're [planning](https://github.com/tarantool/tarantool/issues/7166) to get rid of this fiber function. Perhaps, it's time to add `FIBER_IS_SYSTEM` flag.

Reply:
This sounds reasonable: I made a ticket for that: #7473. I talked to @filonenko-mikhail, he's all right with such feature.

However, I think this patch should be merged too as it's definitely a bug. We should not set `error_reconnect` state or wait if fiber is dead. It's not a way it used to work before and it's not quite logical.

Reply:
The bug will be fixed by marking the fiber system. Internally, a worker fiber can be cancelled only after changing the state machine state to `closed` so it can't possibly set `error_reconnect` in this case. I don't see any point in complicating the code to fix a bug that can be fixed just by marking the net.box worker fiber system.

Reply:
Done. Let's discuss the API of this feature more precisly